scripts/us_census/pep/monthly_population_estimate/download.py

```diff
@@ -0,0 +1,226 @@
+    provided json file and save it as an xlsx file.
+"""
+
+import os
+import json
+import pandas as pd
+import numpy as np
+from absl import app
+from absl import flags
+
+_FLAGS = flags.FLAGS
+_URLS_JSON_PATH = os.path.dirname(os.path.abspath(__file__)) \
+                    + os.sep +"file_urls.json"
+
+_URLS_JSON = None
+with open(_URLS_JSON_PATH, encoding="UTF-8") as file:
+    _URLS_JSON = json.load(file)
+
+# Flag names are globally defined!  So in general, we need to be
+# careful to pick names that are unlikely to be used by other libraries.
+# If there is a conflict, we'll get an error at import time.
+flags.DEFINE_list("us_census_pep_monthly_pop_estimate_url", \
+    _URLS_JSON["urls"], "Import Data URL's List")
+
+_HEADER = 1
+SKIP_ROWS = 1
+_SCALING_FACTOR_TXT_FILE = 1000
+
+
+def _save_data(url: str, download_local_path: str) -> None:
+    """
+    This method loads the Data from url to pandas Dataframe.
+    Writes the data to local path provided as one the parameter.
+
+    Arguments:
+        url (str): Url of the dataset
+        download_local_path (str): LocalPath to save the datasets.
+
+    Returns:
+        None
+    """
+    df = None
+    file_name = url.split("/")[-1]
+    if ".xls" in url:
+        df = pd.read_excel(url, header=_HEADER)
+        df.to_excel(download_local_path + os.sep + file_name,
+                    index=False,
+                    header=False,
+                    engine='xlsxwriter')
+    elif ".csv" in url:
+        file_name = file_name.replace(".csv", ".xlsx")
+        df = pd.read_csv(url, header=None)
+        df = _clean_csv_file(df)
+        df.to_excel(download_local_path + os.sep + file_name,
+                    index=False,
+                    engine='xlsxwriter')
+    elif ".txt" in url:
+        file_name = file_name.replace(".txt", ".xlsx")
+        cols = [
+            "Year and Month", "Date", "Resident Population",
+            "Resident Population Plus Armed Forces Overseas",
+            "Civilian Population", "Civilian NonInstitutionalized Population"
+        ]
+        df = pd.read_table(url,
+                           index_col=False,
+                           delim_whitespace=True,
+                           engine='python',
+                           skiprows=17,
+                           names=cols)
+        df = _clean_txt_file(df, _SCALING_FACTOR_TXT_FILE)
+        df.to_excel(download_local_path + os.sep + file_name,
+                    index=False,
+                    engine='xlsxwriter')
+
+
+def _sum_cols(col: pd.Series) -> pd.Series:
+    """
+    This method concats two DataFrame column values
+    with space in-between.
+
+    Arguments:
+        col[0] (Series) : DataFrame Column of dtype str
+        col[1] (Series) : DataFrame Column of dtype str
+
+    Returns:
+        res (Series) : Concatenated DataFrame Columns
+    """
+    print(type(col))
+    res = col[0]
+    if col[1] is None:
```

Comment:
Similarly, what if col[0] is None?

Reply:
Looking at the data whenever col[0] has year, col[1] is None 
  Thus concatinating Date with Month which is needed here, comment has been mentioned in the code as well